http/endpoints/user/endpoints.go

```diff
@@ -0,0 +1,96 @@
+package userendpoints
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+	userservice "github.com/jumaroar-globant/go-bootcamp/http/service/user"
+	"github.com/jumaroar-globant/go-bootcamp/shared"
+)
+
+//UserEndpoints are the user endpoints
+type UserEndpoints struct {
+	Authenticate endpoint.Endpoint
+	CreateUser   endpoint.Endpoint
+	GetUser      endpoint.Endpoint
+	UpdateUser   endpoint.Endpoint
+	DeleteUser   endpoint.Endpoint
+}
+
+//AuthenticationRequest is the authentication request
+type AuthenticationRequest struct {
+	Username string
+	Password string
+}
+
+//AuthenticationResponse is the authentication response
+type AuthenticationResponse struct {
+	Message string
+}
+
+//GetUserRequest is the get user request
+type GetUserRequest struct {
+	UserID string
+}
+
+//DeleteUserRequest is the delete user request
+type DeleteUserRequest struct {
+	UserID string
+}
+
+//DeleteUserResponse is the delete user response
+type DeleteUserResponse struct {
+	Message string
+}
+
+//MakeEndpoints creates the user endpoints
+func MakeEndpoints(s userservice.Service) *UserEndpoints {
+	return &UserEndpoints{
+		Authenticate: makeAuthenticationEndpoint(s),
+		CreateUser:   makeCreateUserEndpoint(s),
+		GetUser:      makeGetUserEndpoint(s),
+		UpdateUser:   makeUpdateUserEndpoint(s),
+		DeleteUser:   makeDeleteUserEndpoint(s),
+	}
+}
+
+func makeAuthenticationEndpoint(s userservice.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(AuthenticationRequest)
```

Comment:
what happens if the request is not type of AuthenticationRequest

Reply:
Done, added some validations for said cases